cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/JavaSparkFacade.java

```diff
@@ -42,9 +42,8 @@ public JavaSparkFacade(SparkConf sparkConf) {
   @Override
   public <R, K, V> R createRDD(Class<? extends InputFormat> inputFormatClass,
                                Class<K> keyClass, Class<V> valueClass, Configuration hConf) {
-    Configuration configuration = new Configuration(hConf);
-    configuration.set(MRJobConfig.INPUT_FORMAT_CLASS_ATTR, inputFormatClass.getName());
-    return (R) sparkContext.newAPIHadoopRDD(configuration, inputFormatClass, keyClass, valueClass);
+    hConf.set(MRJobConfig.INPUT_FORMAT_CLASS_ATTR, inputFormatClass.getName());
```

Comment:
this - and the following - is what we discussed in person, @chtyim - to avoid double copying.